exchange/bidder_validate_bids.go

```diff
@@ -34,34 +34,41 @@ type validatedBidder struct {
 func (v *validatedBidder) requestBid(ctx context.Context, bidderRequest BidderRequest, conversions currency.Conversions, reqInfo *adapters.ExtraRequestInfo, adsCertSigner adscert.Signer, bidRequestOptions bidRequestOptions, alternateBidderCodes openrtb_ext.ExtAlternateBidderCodes, hookExecutor hookexecution.StageExecutor, ruleToAdjustments openrtb_ext.AdjustmentsByDealID) ([]*entities.PbsOrtbSeatBid, extraBidderRespInfo, []error) {
 	seatBids, extraBidderRespInfo, errs := v.bidder.requestBid(ctx, bidderRequest, conversions, reqInfo, adsCertSigner, bidRequestOptions, alternateBidderCodes, hookExecutor, ruleToAdjustments)
 	for _, seatBid := range seatBids {
-		if validationErrors := removeInvalidBids(bidderRequest.BidRequest, seatBid, bidRequestOptions.responseDebugAllowed); len(validationErrors) > 0 {
+		if validationErrors := removeInvalidBids(bidderRequest.BidRequest, seatBid, &extraBidderRespInfo.adapterNonBids, bidRequestOptions.responseDebugAllowed); len(validationErrors) > 0 {
 			errs = append(errs, validationErrors...)
 		}
 	}
 	return seatBids, extraBidderRespInfo, errs
 }
 
 // validateBids will run some validation checks on the returned bids and excise any invalid bids
-func removeInvalidBids(request *openrtb2.BidRequest, seatBid *entities.PbsOrtbSeatBid, debug bool) []error {
+func removeInvalidBids(request *openrtb2.BidRequest, seatBid *entities.PbsOrtbSeatBid, seatNonBids *nonBids, debug bool) []error {
 	// Exit early if there is nothing to do.
 	if seatBid == nil || len(seatBid.Bids) == 0 {
 		return nil
 	}
 
 	// By design, default currency is USD.
 	if cerr := validateCurrency(request.Cur, seatBid.Currency); cerr != nil {
+		for _, bid := range seatBid.Bids {
+			seatNonBids.addBid(bid, int(ResponseRejectedGeneral), seatBid.Seat)
+		}
 		seatBid.Bids = nil
 		return []error{cerr}
 	}
 
 	errs := make([]error, 0, len(seatBid.Bids))
 	validBids := make([]*entities.PbsOrtbBid, 0, len(seatBid.Bids))
 	for _, bid := range seatBid.Bids {
-		if ok, err := validateBid(bid, debug); ok {
+		ok, err := validateBid(bid, debug)
+		if ok {
 			validBids = append(validBids, bid)
-		} else if err != nil {
+			continue
+		}
+		if err != nil {
 			errs = append(errs, err)
 		}
```

Comment:
@ashishshinde-pubm : Do you see any challenges with following code snippet?
```go
if ok, err := validateBid(bid, debug); ok {
   validBids = append(validBids, bid)
} else if err != nil {
   errs = append(errs, err)
   seatNonBids.addBid(bid, int(ResponseRejectedGeneral), seatBid.Seat) // report seat non bid
}
```

Reply:
@ShriprasadM  when `bid.Bid.Price < 0.0`  , `validateBid` function returns `ok=false` and  `err=nil`
In this case, we will miss to generate nonBid

Reply:
@ashishshinde-pubm :  then instead of checking `nil` on `err`.  how do you see following
```go
if ok, err := validateBid(bid, debug); ok {
   validBids = append(validBids, bid)
}  else {
    seatNonBids.addBid(bid, int(ResponseRejectedGeneral), seatBid.Seat) // report seat non bid
    if err != nil {
        errs = append(errs, err)
    }
}
```

Reply:
done